hermes/crates/cbork/cddl-parser/src/lib.rs

```diff
@@ -45,35 +107,52 @@ pub const POSTLUDE: &str = include_str!("grammar/postlude.cddl");
 /// # Examples
 ///
 /// ```rs
-/// use cddl_parser::parse_cddl;
+/// use cddl_parser::{parse_cddl, Extension};
 /// use std:fs;
 ///
 /// let input = fs::read_to_string("path/to/your/file.cddl").unwrap();
-/// let result = parse_cddl(&input);
+/// let result = parse_cddl(&input, &Extension::CDDLParser);
 /// assert!(result.is_ok());
 /// ```
-pub fn parse_cddl(input: &str) -> Result<(), Box<CDDLError>> {
-    let result = CDDLParser::parse(Rule::cddl, input);
-
-    match result {
-        Ok(c) => println!("{c:?}"),
-        Err(e) => {
-            println!("{e:?}");
-            println!("{e}");
-            return Err(Box::new(CDDLError::from(e)));
+pub fn parse_cddl<'a>(
+    input: &'a str, extension: &Extension,
+) -> Result<Box<AST<'a>>, Box<CDDLError>> {
+    let result = match extension {
+        Extension::RFC8610Parser => {
+            rfc_8610::RFC8610Parser::parse(rfc_8610::Rule::cddl, &input)
+                .map(AST::RFC8610)
+                .map_err(CDDLErrorType::RFC8610)
         },
-    }
+        Extension::RFC9615Parser => {
+            rfc_9615::RFC8610Parser::parse(rfc_9615::Rule::cddl, &input)
+                .map(AST::RFC9615)
+                .map_err(CDDLErrorType::RFC9615)
+        },
+        Extension::CDDLParser => {
+            cddl::RFC8610Parser::parse(cddl::Rule::cddl, &input)
+                .map(AST::CDDL)
+                .map_err(CDDLErrorType::CDDL)
+        },
+    };
+
+    // TODO: parse POSTLUDE and then append into the parsed structure,
+    // as trying to concat the input with `POSTLUDE` first causing a lifetime problem.
+
+    result.map(|ast| Box::new(ast)).map_err(|e| {
+        println!("{e:?}");
+        println!("{e}");
 
-    Ok(())
+        Box::new(CDDLError::from(e))
+    })
 }
 
 #[cfg(test)]
 mod tests {
-    use crate::{parse_cddl, POSTLUDE};
+    use crate::*;
 
     #[test]
     fn it_works() {
-        let result = parse_cddl(POSTLUDE);
+        let result = parse_cddl(POSTLUDE, &Extension::CDDLParser);
```

Comment:
When `POSTLUDE` is automatically included by the parser, this test can just check an empty string.